### PR TITLE
Add enable_dataset_name_normalization option

### PR DESCRIPTION
**Pull request by VioletM**
Now there is an option to control the dataset name normalization separately. `enable_dataset_name_normalization=true`

### CODE REVIEW

File: dlt/common/destination/reference.py [hunk 1]
```diff
@@ -190,6 +190,8 @@ class DestinationClientDwhConfiguration(DestinationClientConfiguration):
     """How to handle replace disposition for this destination, can be classic or staging"""
     staging_dataset_name_layout: str = "%s_staging"
     """Layout for staging dataset, where %s is replaced with dataset name. placeholder is optional"""
+    dataset_name_normalization: bool = True
```
Comment (sh-rp) [inline dlt/common/destination/reference.py:193]: I think an imperative makes more sense her "normalize_dataset_name"

**Review by rudolfix — COMMENTED** [review]
thanks for the tests and improved docs! code LGTM! I agree with Dave's comment to rename the config field

File: dlt/common/destination/reference.py (hunk 1 above)
Comment (VioletM) [reply dlt/common/destination/reference.py:193]: yes, but we already have this field, and I don't want to change it in all places
```
def normalize_dataset_name(self, schema: Schema) -> str:
```

I was thinking of `to_normalize_dataset_name`, but we don't use such a naming in `dlt`. Do you have any ideas for naming?

Comment (rudolfix) [reply dlt/common/destination/reference.py:193]: ok then `enable_dataset_name_normalizaton`?

**Review by rudolfix — APPROVED** [review]
LGTM! thanks for docs and tests

**Pull request merged**